project_remaining_hours_update/models/project_task_remaining_hours.py

```diff
@@ -0,0 +1,43 @@
+# © 2019 Numigi (tm) and all its contributors (https://bit.ly/numigiens)
+# License LGPL-3.0 or later (http://www.gnu.org/licenses/lgpl).
+
+from odoo import fields, models
+
+
+class TaskRemainingHours(models.Model):
+
+    _name = 'project.task.remaining.hours'
+    _description = 'Remaining Hours History on Tasks'
+    _order = 'id desc'
+
+    project_id = fields.Many2one('project.project', related='task_id.project_id', store=True)
+    task_id = fields.Many2one('project.task', required=True, index=True)
+    user_id = fields.Many2one('res.users', readonly=True)
+    planned_hours = fields.Float()
+    total_hours_spent = fields.Float()
+    remaining_hours = fields.Float()
+    remaining_hours_change = fields.Float()
+    comment = fields.Text()
+
+    def _get_remaining_hours_vals(self, task, new_remaining_hours, user, comment):
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Method could be a function](https://app.codacy.com/app/numigi/odoo-project-addons/pullRequest?prid=4293429)